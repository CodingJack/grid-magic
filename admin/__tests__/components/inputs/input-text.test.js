/**
 * External dependencies.
 */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

/**
 * Internal dependencies.
 */
import InputText from '../../../src/js/components/inputs/input-text';

const namespace = 'gmagic';
Enzyme.configure( { adapter: new Adapter() } );


test( 'it renders correctly', () => {
  const tree = create(
    <InputText 
      value="Some Value"
      ariaLabel="Some Label"
      namespace={ namespace } 
      onChange={ () => {} }
    />
  ).toJSON();
  
  if( ! tree ) {
    return;
  }
  expect( tree ).toMatchSnapshot();
  const { props } = tree;
  
  props.onChange = () => {};
  props.namespace = 'name-space';
  props.value = 'Another Value';
  props.ariaLabel = 'Another Label';
  props.extraClass = 'my-extra-class';
  props.placeholder = 'some placeholder';
  props.readonly = true;

  expect( tree ).toMatchSnapshot();
} );

describe('it unmounts', () => {
  const onChange = jest.fn();
  
  const component = shallow(
    <InputText 
      value="Some Value" 
      ariaLabel="Some Label"
      namespace={ namespace } 
      onChange={ onChange }
    /> 
  );
  
  test( 'Change event', () => {
    component.simulate( 'change', {
      target: {
        value: 'new value',
      },
    } );
    expect( onChange ).toHaveBeenCalled();
  } );

  it('should match the snapshot', () => {
    expect( component.html() ).toMatchSnapshot();
  });

  test( 'it unmounts', () => {
    component.unmount();
  } );
} );




