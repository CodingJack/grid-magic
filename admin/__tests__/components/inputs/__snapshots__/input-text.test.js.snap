// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<input
  aria-labelledby="Some Label"
  className="esg-v4-input-text"
  onChange={[Function]}
  placeholder=""
  readOnly={false}
  tabIndex="0"
  type="text"
  value="Some Value"
/>
`;

exports[`it renders correctly 2`] = `
<input
  aria-labelledby="Some Label"
  ariaLabel="Another Label"
  className="esg-v4-input-text"
  extraClass="my-extra-class"
  namespace="name-space"
  onChange={[Function]}
  placeholder="some placeholder"
  readOnly={false}
  readonly={true}
  tabIndex="0"
  type="text"
  value="Another Value"
/>
`;

exports[`it unmounts should match the snapshot 1`] = `"<input type=\\"text\\" class=\\"esg-v4-input-text\\" value=\\"Some Value\\" placeholder=\\"\\" tabindex=\\"0\\" aria-labelledby=\\"Some Label\\"/>"`;
